### DISCUSSION TITLE
Auto-completing of code blocks

### DISCUSSION THREAD

**Discussion by thomprl** [codefori/vscode-ibmi#1289] [General]
I'm wondering if I missed something but coming from RDi and Visual Studio .NET, another feature I'm missing is the auto complete of the IF blocks.  Meaning if I type an IF statement and press enter it auto adds the ENDIF; at the end of the block.  Same with DCL-PROC/END-PROC.  Is this available in VSCode?  Just a time saver.

**Comment by worksofliam** [comment]
Do you have the RPGLE extension? It is provided by that. 

https://marketplace.visualstudio.com/items?itemName=HalcyonTechLtd.vscode-rpgle

**Reply by thomprl** [reply]
Yes I have it installed, I guess it just doesn't work the way I'd expect it to or I don't know a shortcut.  

In RDi you can type If 1=1; and press enter and the EndIf; is automatically added.  I just saw if you type IF and then enter it will add the EndIf.  Would be nice if it would let you complete your IF statement and then add it so you aren't having to jump back up and finish your IF statement.

**Reply by Gregory1701** [reply]
This isn't working for me either.  I have v0.19.1 of the RPG extension, which I installed via the IBM i Development Pack v0.1.0.  Are these the most current versions?

**Reply by thomprl** [reply]
The only way I can get it to auto complete is just type IF and then press enter.  It will then go ahead and add the ENDIF.  Would be nice if it waited until the end of you type the IF statement or code block and then auto add a ENDIF.  Not sure what is possible in VSCode.  Or type DCL-PROC MyProc; and press enter and have the END-PROC added.

**Reply by chrjorgensen** [reply]
@thomprl @Gregory1701 You could use snippets for this - they are already defined for ILE RPG. You can find them by pressing `F1` and type `snippets`.

I think this is the autocomplete 'philosophy' of VS Code...
